Document customizable binary operators

JeffFessler opened the pull request:
I learned indirectly through comments on a PR that \otimes and \oplus and \odot can be extended as binary operators yet I cannot find this fact anywhere in the documentation.  Search the docs for `oplus` yields nothing.
Somewhere there must be a list of which characters can be extended this way and it should be documented in the manual.
This PR is an attempt to get that ball rolling!

JeffFessler committed:
Document customizable binary operators

I learned indirectly through comments on a PR that \otimes and \oplus and \odot can be extended as binary operators yet I cannot find this fact anywhere in the documentation.  Search the docs for `oplus` yields nothing.
Somewhere there must be a list of which characters can be extended this way and it should be documented in the manual.
This PR is an attempt to get that ball rolling!

mcabbott commented:
Sounds good to have. The complete list is here: 
https://github.com/JuliaLang/julia/blob/master/src/julia-parser.scm#L29

(Note also that by postfix I think you mean prefix!)

I also can’t find where it is mentioned that things like `*′(x,y) = x*y/2` also create infix operators. Is this in the manual?

JeffFessler committed:
Add url for list and fix prefix typo

JeffFessler commented:
@mcabbott thanks for the link.  I added it to the PR and also fixed the typo you mentioned.

I was unaware of the syntax you mentioned.  Cautionary note to anyone who tries to reproduce it: the quote in `*′(x,y) = x*y/2` is not an ordinary apostrophe!  Seems like something that should be documented as well.  Not sure where it would go.

mcabbott commented:
It may be mentioned somewhere logical! But you are allowed any unicode decoration like `⊗ᵣ(x,y) = kron(y,x); /₂(x,y) = x/y/2`. 

The exact line I linked is the symbols parsed like `*` (in terms of precedence, and associativity), but there are more parsed like `+` or like `->` etc.

JeffFessler committed:
Expand docs of infix binary operators.

JeffFessler commented:
Thanks for the clarification.  I've updated the PR to tell the story somewhat more generally.

StefanKarpinski commented:
@JeffBezanson, can you give a read through and merge if you approve?

JeffFessler commented:
@JeffBezanson, it's been a few weeks so I thought I'd make a gentle ping...
